src/gluonts/representation/mean_scaling.py

```diff
@@ -0,0 +1,105 @@
+# permissions and limitations under the License.
+
+from .representation import Representation
+
+import mxnet as mx
+
+
+# Standard library imports
+from typing import Tuple, Optional
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.model.common import Tensor
+
+
+class MeanScaling(Representation):
+    """
+        A class representing a mean scaling approach.
+        Inputs are simply rescaled based on their mean.
+
+        Parameters
+        ----------
+        clip_max
+            The maximum value to which re-scaled values will be clipped to. By default, max values will not be clipped.
+            (default: -1)
+        minimum_scale
+            The minimum value to which re-scaled values will be clipped to.
+            (default: 1e-10)
+    """
+
+    @validated()
+    def __init__(
+        self, clip_max: int = -1, scale_min: float = 1e-10, *args, **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.scale_min = scale_min
+        self.clip_max = clip_max
+        self.means = mx.ndarray.array([])
+
+    def compute_scale(
+        self, F, data: Tensor, observed_indicator: Tensor  # shapes (N, T, C)
+    ) -> Tensor:
+        # these will have shape (N, C)
+        num_observed = F.sum(observed_indicator, axis=1)
+        sum_observed = (data.abs() * observed_indicator).sum(axis=1)
+
+        # first compute a global scale per-dimension
+        total_observed = num_observed.sum(axis=0)
+        denominator = F.maximum(total_observed, 1.0)
+        default_scale = sum_observed.sum(axis=0) / denominator  # shape (C, )
+
+        # then compute a per-item, per-dimension scale
+        denominator = F.maximum(num_observed, 1.0)
+        scale = sum_observed / denominator  # shape (N, C)
+
+        # use per-batch scale when no element is observed
+        # or when the sequence contains only zeros
+        cond = F.broadcast_greater(sum_observed, F.zeros_like(sum_observed))
+        scale = F.where(
+            cond,
+            scale,
+            F.broadcast_mul(default_scale, F.ones_like(num_observed)),
+        )
+
+        return F.maximum(scale, self.scale_min)
+
+    # noinspection PyMethodOverriding
+    def hybrid_forward(
+        self,
+        F,
+        data: Tensor,
+        observed_indicator: Tensor,
+        scale: Optional[Tensor],
+    ) -> Tuple[Tensor, Tensor]:
+        data = F.cast(data, dtype="float32")
+
+        if scale is None:
+            scale = self.compute_scale(F, data, observed_indicator)
+            scale = scale.expand_dims(axis=1)
+        self.means = scale
+
+        scaled_data = F.broadcast_div(data, scale)
+
+        if self.clip_max != -1:
+            scaled_data = F.clip(scaled_data, -self.clip_max, self.clip_max)
+
+        if self.is_output:
+            return scaled_data, scale
+        else:
+            return F.expand_dims(scaled_data, axis=1), scale
```

Comment:
I have generalized the models (`WaveNet`, `DeepAR`, `Transformer`, `SimpleFeedForward`) to deal with 3-dimensional tensors. Since binnings require a three dimensional tensor of shape `(batch_size, embedding_size, pred_length)`, I have also adapted the mean scaler to output `(batch_size, 1, pred_length)`-shaped tensors. This prevents us from having to handle different dimensions in the models and abstracts this into the representation. However, tensor expansion is only required in input mode, and not in output mode. See larger comment below for more on this.